tethysext/atcore/models/file_database/file_database_client.py

```diff
@@ -71,3 +77,78 @@ def path(self, root_dir: str) -> None:
         root_dir (str): the directory to be the root directory of the file database.
         """
         self._path = os.path.join(root_dir, str(self.instance.id))
+
+    def get_collection(self, collection_id: uuid.UUID) -> FileCollectionClient:
+        """
+        Get a FileCollectionClient owned by this FileDatabase by its collection_id
+
+        Args:
+            collection_id (uuid.UUID): The id for the file collection owned by this FileDatabase.
+
+        Returns:
+            The FileCollectionClient for the FileCollection.
+        """
+        collection_query = self._session.query(FileCollection).filter_by(
+            id=collection_id, file_database_id=self.instance.id
+        )
```

Comment:
Add `.one()` to the end of this and then catch the exception if MultipleResultFound or NoResultFound is raised and throw our own exception: https://stackoverflow.com/questions/18110033/getting-first-row-from-sqlalchemy

Reply:
Okay i cought the NoResultFound. I didn't put the MultipleResultFound in because i don't think it is possible. We are filtering by an ID which means i should never get more than one, and If i do it should have been caught well before here.